api/v1alpha1/releaseserviceconfig_types.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const ReleaseServiceConfigResourceName string = "config"
+
+// ReleaseServiceConfigSpec defines the desired state of ReleaseServiceConfig.
+type ReleaseServiceConfigSpec struct {
+	// AllowLocalPipelines is the boolean that specifies whether or not pipelines stored locally
+	// on the cluster may be used
+	// +optional
+	AllowLocalPipelines bool `json:"allowLocalPipelines,omitempty"`
+}
+
+// ReleaseServiceConfigStatus defines the observed state of ReleaseServiceConfig.
+type ReleaseServiceConfigStatus struct {
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// ReleaseServiceConfig is the Schema for the releaseserviceconfigs API
+type ReleaseServiceConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ReleaseServiceConfigSpec   `json:"spec,omitempty"`
+	Status ReleaseServiceConfigStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// ReleaseServiceConfigList contains a list of ReleaseServiceConfig
+type ReleaseServiceConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ReleaseServiceConfig `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&ReleaseServiceConfig{}, &ReleaseServiceConfigList{})
+}
+
+// IsBundleLocalAndAllowed checks whether the bundle value is empty if AllowLocalPipelines is False.
+func (r *ReleaseServiceConfig) IsBundleLocalAndAllowed(bundle string) bool {
```

Comment:
Probably cleaner to just access the values directly although I don't oppose. If you don't go for the `debug` field name, this function should be renamed to something better. Maybe something like `localPipelinesAllowed` for readability or `AreLocalPipelinesAllowed` to start with  `Is`/`Are` like in the Release type.

Reply:
Okay, I switched (not pushed yet) `AllowLocalPipelines` to `Debug`. What should I name this then? Should I have a debug check function with a TBD name that calls this? Since there will be more debug checks than just that? What are your thoughts

Reply:
@davidmogar

Reply:
In the internal services repo we access the `Debug` field directly without a method.  Regarding whether we will have more debug checks... Not sure... Probably not. And even if we have them, maybe the `Debug mode` should be consistent and just to a set of things always. In other words, not a configurable `Debug mode`. Wdyt?

Reply:
I would expect debug mode to ignore the pipelineAllowList too, so it may have more checks. I am not sure I follow the second part of your response about it being a set of things. As in, if debug == true, you get these things (like local bundles allowed, pipelineAllowList ignored), if debug == false, production behvaior. no other option?

Reply:
Yeah, that's what I meant. Reason for this is that the AllowLocalPipelines is a debug thing. It's not something we will offer to users.

Reply:
modified